File: shared/schemas/feedback.ts
export type FeedbackRating = "thumbs_up" | "thumbs_down";

export type FeedbackRequest = {
  message_id: number;
  rating: FeedbackRating;
  tags?: string[];
  rewrite_text?: string;
};

export type FeedbackResponse = {
  status: "ok" | "error";
};

File: apps/chatbot/backend/app/schemas/feedback.py
from pydantic import BaseModel, Field
from typing import List, Optional


class FeedbackRequest(BaseModel):
    message_id: int
    rating: str = Field(..., pattern="^(thumbs_up|thumbs_down)$")
    tags: Optional[List[str]] = None
    rewrite_text: Optional[str] = None


class FeedbackResponse(BaseModel):
    status: str = "ok"
